pkg/configmap/configmap.go

```diff
@@ -50,17 +53,32 @@ func (l *BundleLoader) Load(cm *corev1.ConfigMap) (bundle *api.Bundle, err error
 	return
 }
 
-func loadBundle(entry *logrus.Entry, data map[string]string) (bundle *api.Bundle, skipped map[string]string, err error) {
+func loadBundle(entry *logrus.Entry, cm *corev1.ConfigMap) (bundle *api.Bundle, skipped map[string]string, err error) {
 	bundle = &api.Bundle{Object: []string{}}
 	skipped = map[string]string{}
 
+	gzipped := hasGzipEncodingAnnotation(cm)
+	if gzipped {
+		entry.Info("Decoding gzip-encoded bundle data")
+	}
+
 	// Add kube resources to the bundle.
-	for name, content := range data {
-		reader := strings.NewReader(content)
+	for name, content := range cm.Data {
 		logger := entry.WithFields(logrus.Fields{
 			"key": name,
 		})
 
+		var reader io.Reader
+		if gzipped {
+			decoded, err :=  encoding.DecodeGzipBase64([]byte(content))
+			if err != nil {
+				return nil, nil, fmt.Errorf("error decoding gzip-encoded bundle data: %v", err)
+			}
+			reader = bytes.NewReader(decoded)
```

Comment:
The `reader` is only used by the `registry.DecodeUnstructured(reader)` below, but the actual decompressed `content []byte` is still needed further down (`bundle.Object = append(bundle.Object, content)`). But since we've given up on our own base64 encoding, this clearly uses less memory now.